distributed/shuffle/tests/test_graph.py

```diff
@@ -29,6 +29,12 @@ def test_basic(client):
     # ^ NOTE: this works because `assert_eq` sorts the rows before comparing
 
 
+def test_raise_on_non_string_column_name():
+    df = dd.from_pandas(pd.DataFrame({"a": range(10), 1: range(10)}), npartitions=5)
+    with pytest.raises(TypeError, match="p2p requires all column names to be str"):
+        df.shuffle("x", shuffle="p2p")
```

Comment:
nit: this would be an invalid operation anyhow since `x` doesn't exist. this test implicitly assumes that the one check happens before the other one.

```suggestion
        df.shuffle("a", shuffle="p2p")
```